packages/design-system/src/components/Autocomplete/utils.tsx

```diff
@@ -89,3 +89,28 @@ export function removeUndefined<T>(obj: T): T {
   Object.keys(obj).forEach((key) => obj[key] === undefined && delete obj[key]);
   return obj;
 }
+
+export const isGroup = (
+  item: AutocompleteItem | AutocompleteItemGroup
+): item is AutocompleteItemGroup => {
+  return (
+    typeof (item as AutocompleteItemGroup).label === 'string' &&
+    Array.isArray((item as AutocompleteItemGroup).items)
+  );
+};
+
+export const findItemById = (
+  id: string,
+  items: AutocompleteItems
+): AutocompleteItem | undefined => {
+  for (const item of items) {
+    if (isGroup(item)) {
```

Comment:
I think it's possible to simplify this logic and remove the `isGroup` function if we swap this out to:
```suggestion
    if ('items' in item) {
```
since `items` only exists on the AutocompleteItemGroup type